tests/Struct/FindologicProductTest.php

```diff
@@ -379,6 +379,72 @@ public function testProductRatings(array $ratings, float $expectedRating): void
         $this->assertEquals($expectedRating, current($ratingAttribute->getValues()));
     }
 
+    public function attributeProvider(): array
+    {
+        return [
+            'filter with some special characters' => [
+                'attributeName' => 'Special Characters /#+*()()=§(=\'\'!!"$.|',
+                'expectedName' => 'SpecialCharacters'
+            ],
+            'filter with brackets' => [
+                'attributeName' => 'Farbwiedergabe (Ra/CRI)',
+                'expectedName' => 'FarbwiedergabeRaCRI'
+            ],
+            'filter with special UTF-8 characters' => [
+                'attributeName' => 'Ausschnitt D ø (mm)',
+                'expectedName' => 'AusschnittDmm'
+            ],
+            'filter dots and dashes' => [
+                'attributeName' => 'free_shipping.. Really Cool--__',
+                'expectedName' => 'free_shippingReallyCool--__'
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider attributeProvider
+     */
+    public function testAttributesAreProperlyEscaped(string $attributeName, string $expectedName): void
+    {
+        $productEntity = $this->createTestProduct([
+            'properties' => [
+                [
+                    'id' => Uuid::randomHex(),
+                    'name' => 'some value',
+                    'group' => [
+                        'id' => Uuid::randomHex(),
+                        'name' => $attributeName
+                    ],
+                ]
+            ]
+        ]);
+
+        $criteria = new Criteria([$productEntity->getId()]);
+        $criteria = Utils::addProductAssociations($criteria);
```

Comment:
Just curious, what are all those associations which get added in this method?

Reply:
Associations are basically database joins. For products we have several like, images, properties, etc.